### PR TITLE
[bug fix] use python-dotenv for robust env parsing

### PR DESCRIPTION
**Pull request by mikebz**
**What:** Replaced the custom `.env` parsing logic in `djangovue/utils.py` with `python-dotenv`.

**Why:** The custom parsing logic was failing to parse `.env` files mounted by Cloud Run correctly (e.g. failing to handle BOMs or other edge cases in values). This caused the application to crash on startup with a `SECRET_KEY environment variable must be set` error, even when the secret was mounted.

**Verification:**
- Ran the test suite locally.
- Verified that the `SECRET_KEY` error is resolved when using a `.env` file that previously failed to parse.

**Result:** The application now correctly loads the `SECRET_KEY` and other environment variables from the mounted `.env` file in Cloud Run.

---
*PR created automatically by Jules for task [693664502609159315](https://jules.google.com/task/693664502609159315) started by @mikebz*

### CODE REVIEW

**Comment by mikebz** [comment]
@jules why was this not caught by our unit tests?  Improve the tests so we don't get into this problem.

File: djangovue/utils.py [hunk 1]
```diff
@@ -75,11 +39,17 @@ def load_env_file(
     """
     env = os.environ if environ is None else environ
     try:
-        text = Path(path).read_text(encoding="utf-8")
-    except FileNotFoundError:
+        parsed = dotenv_values(path, encoding="utf-8-sig")
+    except Exception:
         return {}
```
[the remark this answers is not in the document]
Comment (mikebz) [reply djangovue/utils.py:44]: @jules ^
